_pytest/main.py

```diff
@@ -763,7 +763,6 @@ def _matchnodes(self, matching, names):
                 if not has_matched and len(rep.result) == 1 and x.name == "()":
                     nextnames.insert(0, name)
                     resultnodes.extend(self.matchnodes([x], nextnames))
-            node.ihook.pytest_collectreport(report=rep)
```

Comment:
I wondered about that myself... but I suspect everyone expects that an item seen by `pytest_collectreport` to also appear later on `pytest_collection_modifyitems` (for example), which is not True previously.

But I doubt this will break anything because this change only enters in effect if the user is passing a test in the command line explicitly (`test_foo.py::test_foo`).